packages/dso-toolkit/gulp/build-svg-spritesheets.js

```diff
@@ -0,0 +1,165 @@
+            const variants = ast.stylesheet.rules
+              .filter((rule) => rule.type === "rule") // Keep only CSS rules (ignore comments, etc.)
+              .flatMap((rule) => {
+                const colorDeclaration = rule.declarations.find((d) => d.property === "color");
+                if (!colorDeclaration) return [];
+
+                const color = colorDeclaration.value;
+                return rule.selectors
+                  .map((selector) => {
+                    if (selector === id) {
+                      return { selector: "default", color }; // Mark as default variant
+                    }
+
+                    if (selector.startsWith(`${id}:`)) {
+                      return { selector: selector.slice(id.length + 1), color }; // Extract the variant name
+                    }
+
+                    return null;
+                  })
+                  .filter(Boolean); // Remove null entries
+              });
+
+            return { id, variants };
+          });
+
+          resolve(parsedStyles);
+        }),
+      );
+  });
+}
+
+async function buildDiSpritesheet() {
+  const iconStylesheets = await parseStylesheets(); // Parse stylesheets within the function
+
+  if (!iconStylesheets || iconStylesheets.length === 0) {
+    throw new Error("No icon stylesheets found. Pipeline terminated.");
+  }
+
+  return gulp
+    .src(`${iconsPath}/*.svg`)
+    .pipe(
+      svgstore({
+        inlineSvg: true,
+      }),
+    )
+    .pipe(
+      cheerio({
+        run($) {
+          const symbols = $("symbol").toArray();
+          let position = 0;
+
+          // Create a map for quick stylesheet lookup
+          const stylesheetMap = Object.fromEntries(iconStylesheets.map((s) => [s.id, s]));
+
+          for (const symbol of symbols) {
+            const $symbol = $(symbol);
+            const id = $symbol.attr("id");
+            const stylesheet = stylesheetMap[id];
+
+            if (!stylesheet) {
+              $symbol.remove();
+              continue;
+            }
+
+            const variants = [
+              { selector: id, color: stylesheet.variants.find((v) => v.selector === "default")?.color },
+              ...stylesheet.variants
+                .filter((v) => v.selector !== "default")
+                .map((v) => ({ selector: `${id}-${v.selector}`, color: v.color })),
+            ];
+
+            for (const [index, variant] of variants.entries()) {
+              const x = (position + index) * (canvasSize + gutterSize) + gutterSize / 2;
+
+              const view = $("<view>")
+                .attr("id", `${variant.selector}`)
+                .attr("viewBox", [x, 0, canvasSize, canvasSize].join(" "));
+
+              const svgElement = $("<svg>").attr("id", variant.selector).attr("viewBox", $symbol.attr("viewBox"));
+
+              // Convert children to an array and clone each child
+              for (const child of $symbol.children().toArray()) {
```

Comment:
Het duurde even voordat ik begreep waar dit voor nodig is. Er wordt een nieuw `<svg>` element gemaakt, en de inhoud uit het oude `<symbol>` element wordt gecloned (eventueel voorzien van kleur) en toegevoegd in het nieuwe `<svg>` element.

Het nadeel van deze opzet is:
- Het nieuwe `<svg>` element krijgt alleen de attributen die in deze routines worden gezet. Als een bron `<svg>` [extra attributen](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute) krijgt worden deze niet meegenomen. We maken een spritesheet, ik denk dat onze routines attributen moeten opt-outen (zoals id) ipv moeten opt-innen (zoals nu)
- `.children()` neemt geen whitespace of textnodes mee, mocht de auteur dat in de bron `<svg>` verwerken komt dat niet in de svg spritesheet.